services/provider/common/mock.go

```diff
@@ -0,0 +1,95 @@
+package common
+
+import "os"
+
+type MockError string
+
+const (
+	// Onboarding
+	OnboardInternalError MockError = "ONBOARD_INTERNAL_ERROR"
+	OnboardInvalidToken  MockError = "ONBOARD_INVALID_TOKEN"
+	OnboardInvalidArg    MockError = "ONBOARD_INVALID_ARG"
+
+	// StorageConfig
+	StorageConfigInternalError    MockError = "STORAGE_CONFIG_INTERNAL_ERROR"
+	StorageConfigInvalidUID       MockError = "STORAGE_CONFIG_INVALID_UID"
+	StorageConfigConsumerNotReady MockError = "STORAGE_CONFIG_CONSUMER_NOT_READY"
+
+	// Offboard
+	OffboardInternalError    MockError = "OFFBOARD_INTERNAL_ERROR"
+	OffboardInvalidUID       MockError = "OFFBOARD_INVALID_UID"
+	OffBoardConsumerNotFound MockError = "OFFBOARD_CONSUMER_NOT_FOUND"
+
+	// Update
+	UpdateInternalError    MockError = "UPDATE_INTERNAL_ERROR"
+	UpdateInvalidArg       MockError = "UPDATE_INVALID_ARG"
+	UpdateInvalidUID       MockError = "UPDATE_INVALID_UID"
+	UpdateConsumerNotFound MockError = "UPDATE_CONSUMER_NOT_FOUND"
+
+	MockConsumerID = "vMHA0ppPbjg5TlgvMFcaH4QlQEJB68u+1jWQJ9O9xvde8fxz5vBuu2F6bVIY6pAYLVrC3FajrK1KxmhFTzNDow=="
+
+	MockGrantedCapacity = "2Ti"
+)
+
+var (
+	MockProviderAPI      = os.Getenv("MOCK_PROVIDER_API")
```

Comment:
It will be fetched once while starting, which means even if we change this in b/w it won't affect. I would suggest getting this in every call so that it can be changed on the fly before making a call.

Reply:
good point. I'll update this.

Reply:
done.